### ISSUE TITLE
[Global] Icons and borders are not visible in Windows High Contrast mode

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#408]
## Issue summary

Control borders and UI icons are not visible in [Windows High Contrast mode](https://support.microsoft.com/en-us/windows/use-high-contrast-mode-in-windows-10-fedc744c-90ac-69df-aed5-c8a90125e696). This would cause difficulty in identifying pieces of UI, sizing of controls, and their purpose. Low-vision, high-contrast users should have equal access to content and imagery that sighted users have access to.

<details>
  <summary>Screenshots</summary>

<img width="564" alt="Dawn header section, utility section is highlighted, icons are not visible." src="https://user-images.githubusercontent.com/1392632/129605460-2f351ebd-caae-4e30-9bb5-08cee92bbbcc.png">
<img width="577" alt="Dawn product page, product accordion section is highlighted, icons are not visible." src="https://user-images.githubusercontent.com/1392632/129605518-5f430921-973b-48d8-8e25-8039b62eb19f.png">
<img width="1589" alt="Dawn footer section, email signup and social link sections are highlighted, icons are not visible." src="https://user-images.githubusercontent.com/1392632/129605567-65822c02-be2f-4a80-92b5-618c1cf33b19.png">
<img width="845" alt="Dawn login form, inputs are highlighted, borders are not visible." src="https://user-images.githubusercontent.com/1392632/129605657-78c2b24f-860e-4d40-a5ed-acd23b5639ff.png">
<img width="684" alt="Dawn search form, inputs are highlighted, borders and icon button is not visible." src="https://user-images.githubusercontent.com/1392632/129605703-a01a8759-e866-4277-9af0-336704bb32f7.png">
<img width="1585" alt="Dawn cart page, quantity input and remove controls are highlighted, icons and borders are not visible." src="https://user-images.githubusercontent.com/1392632/129605763-90ff5653-cc3b-4606-ad15-03227ee7d485.png">

</details>

<!-- ### Current code

#### HTML

```html

``` -->

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. With a Windows High Contrast theme activated, review the icons and button/input borders

## Behavior

### Expected

- Icons and control borders to be visible.

### Actual

- Icons are difficult to perceive or are not visible. Borders do not convey control sizing or affordance.

## Recommendation

1. In order for icons to be visible in High Contrast themes and not disrupt the intended design, use the [`forced-colors` media query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors) to set the icon color based on the user preference.
2. To provide `button` and `input` sizing and appearance/affordance, apply a `transparent` border. This will allow a border to appear in High Contrast mode and not disrupt the intended design.

**Be sure to test all UI elements; the code snippet below may not cover all instances.**

### Recommended code

#### CSS

```css
@media screen and (forced-colors: active) {
  .icon {
    color: CanvasText;
    fill: CanvasText;
  }
}

.button, .field__input {
  border: solid 1px transparent;
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
- **Severity:** High
- **Effort:** Low
